internal/cloudsdk/region.go

```diff
@@ -421,16 +421,13 @@ func (c *RegionServiceClient) CreatePrivateLinkAwait(ctx context.Context, id uin
 	err = wait.Poll(ctx, func() (bool, error) {
 		link, err := c.GetPrivateLink(ctx, id, info.Id)
 		if err != nil {
+			if errors.Is(err, ErrPrivateLinkNotFound) {
+				return false, nil
+			}
 			return false, err
 		}
 		rtn = link
-		if link.Status == apigen_mgmt.ERROR {
-			return false, errors.Errorf("failed to create private link, id: %s, creation of privateLink fails and enters an ERROR state", info.Id)
-		}
-		if link.Status == apigen_mgmt.CREATED {
-			return true, nil
-		}
-		return false, nil
+		return true, nil
```

Comment:
Sorry, I don't fully understand the explanation in the PR description. Why is the status returned by `GetPrivateLink` not reflecting its real status?

Reply:
so there are two concepts here: the real private link in the cloud platform and the private link object at the control plane side. The private link object is an actual thing in the control plane, it represents the real private link. The cloud users can only access the private link object. 

In the current design, if the underlying real private link resource is not successfully created, the control plane will still keep the private link object, which shows an ERROR in its status field.

the terraform module does not manage the real private link resource, instead, it is manging the private link object at the control plane side. 

If a private link object is in ERROR state, it still presents in the control plane. That means if it is not stored in the terraform state, then there will an inconsistent issue. Because next time you try to create a private link with the same connection name. It will be rejected because there is a conflict.

Reply:
I see, so the bug is that a privatelink in error status won't be store in the TF state at creation.

One follow up question: if the creation failed and result into a privatelink object in error status, in the next plan the privatelink will be recreated (i.e be destroyed first then be created again) right?

Reply:
currently, it remains unchanged. If the arguments are the same. Then the result probably will be the same. 
Let me check if we can store the state at first then throw an error to let users know there is an error.

Reply:
If I create helm_release resource and the creation failed, next time when I apply, terraform will re-create the resource.

I think we can refer to their implementation if the behavior makes sense to you.

This is slightly unrelated to this fix, which should be merged anyway.